src/packages/async/async.cc

```diff
@@ -557,9 +452,8 @@ void f_async_write() {
 
 #ifdef F_ASYNC_GETDIR
 void f_async_getdir() {
-  function_to_call_t *cb = get_cb();
-  process_efun_callback(1, cb, F_ASYNC_READ);
-  cb->f.fp->hdr.ref++;
+  auto *cb = new function_to_call_t;
+  process_efun_callback(1, cb, F_ASYNC_GETDIR);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Skipping configuration 'F_ASYNC_GETDIR' since the value of 'F_ASYNC_GETDIR' is unknown. Use -D if you want to check it. You can use -U to skip it explicitly.](https://app.codacy.com/gh/fluffos/fluffos/pullRequest?prid=5214413)